### ISSUE TITLE
Dexter doesn't support a WiFi adapter

### ISSUE THREAD

**Issue by JamesNewton** [HaddingtonDynamics/Dexter#51]
In the past, Dexter was unable to support a USB WiFi adapter for RF networking, because the driver would not load on the version 12 operating system. 

A new Ubuntu 16.04 image being completed to close #25 might allow a WiFi adapter to be added and work.

There appears to be a special adapter designed for the microzed board:
http://www.zedboard.org/product/wilink-8-adaptor

However, if a USB WiFi adapter is preferred, the current placement of the fan, at the bottom of the microzed board, blocks all the USB connectors (and the sd card). Moving the fan out to the skin and ducting the air to the stepper motor drivers would give us the physical access we need the USB hub for connecting peripherals to Dexter.

**Comment by JamesNewton** [comment]
I think trying a cheap adapter first is a great idea... I'm not sure why they have that specific adapter...

**Comment by JamesNewton** [comment]
Resolved with the upgraded OS #25

**Comment by JamesNewton** [comment]
One issue with WiFi adapters is that the DHCP protocol will assign an IP on the local router and it's difficult to find out which IP was assigned. I remembered that there was a javascript page that could decode morse code:
https://www.bennadel.com/resources/demo/morse_code/ 
https://www.bennadel.com/blog/2267-decoding-morse-code-with-javascript.htm 
 
Take the keyboard events off that and add processing from the camera:
https://googlechrome.github.io/samples/image-capture/grab-frame-take-photo.html 
 
Accessing the pixels in the picture isn't hard, and you can do a threshold to see if the picture is overall bright (LED on) or dark (LED off).
https://github.com/JamesNewton/AdvancedRoboticsWithJavascript/wiki#image-subtraction-in-a-smartphone-browser 
 
Put that on a public web page (must have HTTPS, not HTTP) and then users can go to the web page, allow access to the camera, hold the camera over the robot, get the web URL decoded, and then click on it to connect directly to the robot!

Users don't even need to know that it's using morse code. But it's also sort of cool that it could decode any message.

**Comment by JamesNewton** [comment]
Although the wifi adapter does work, it can't be directly plugged into the USB A connector on the MicroZed board because that places it directly over the stepper drivers which get quite hot, causing the adapters to fail. Instead, a short USB A male to USB A female extension cable or small USB hub should be used to bring the port out into cooler air.

**Comment by JamesNewton** [comment]
For more information, management, etc... see:
https://github.com/HaddingtonDynamics/Dexter/wiki/Dexter-Networking#wifi

**Comment by JamesNewton** [comment]
To avoid the heating problems with the little [Edimax-EW-7811Un](https://www.amazon.com/Edimax-EW-7811Un-150Mbps-Raspberry-Supports/dp/B003MTTJOY)  chipset RTL8188CUS
 ( which seems to overheat because it's entirely enclosed in the USB slot ) I got a new adapter which uses the similar RTl8192cu chipset but has a larger body which extends outside the slot. 
(https://www.adafruit.com/product/1012)
It works fine, but I notice that it printed this message to the console when I plugged it in:
`ftlwifi: Firmware rtlwifi/rtl8192cufw_TMSC.bin not available`
And yet it works...

I also tried this one but it doesn't work.
https://www.amazon.com/Panda-300Mbps-Wireless-USB-Adapter/dp/B00EQT0YK2/   
And I about drove myself nuts over the last day and a half trying to get the driver to compile for it.
